### DISCUSSION TITLE
Support optional OIDC support for the Kakao provider

### DISCUSSION THREAD

**Discussion by sean-ahn** [nextauthjs/next-auth#8107] [Ideas]
The OAuth provider **Kakao**, widely used in South Korea, offers optional(opt-in) support for OpenID Connect.

To use this feature, I have updated my `AuthOptions` with the following code for the `KakaoProvider`:
```diff
import KakaoProvider from 'next-auth/providers/kakao'

...

    KakaoProvider({
      clientId: env.KAKAO_OAUTH_CLIENT_ID,
      clientSecret: env.KAKAO_OAUTH_CLIENT_SECRET,
+     wellKnown: 'https://kauth.kakao.com/.well-known/openid-configuration',
+     authorization: { params: { scope: 'openid' } },
+     idToken: true,
+     profile(profile) {
+       return {
+         id: profile.sub,
+         name: profile.nickname,
+         email: profile.email,
+         image: profile.picture,
+       }
+     },
    }),

...
```

It works well and I am considering contributing to the project to make Kakao provider support OIDC by default.
However, this OIDC feature is not well-known and is not well-documented.

To address this, I propose adding another provider function called `KakaoOIDC` that users can explicitly import when needed like: 
```diff
- import KakaoProvider from 'next-auth/providers/kakao'
+ import { KakaoOIDC as KakaoProvider } from 'next-auth/providers/kakao' 
```

Before contributing, I would like to ask the code owners about best practices for this case. Please let me know if there are any precedents for this use case. 🙂 

Official documentation (in English)
* https://developers.kakao.com/docs/latest/en/kakaologin/rest-api#request-token
* https://developers.kakao.com/docs/latest/en/kakaologin/rest-api#get-discovery

Related discussions
* https://github.com/nextauthjs/next-auth/discussions/7283

**Comment by JungHoonGhae** [comment]
@sean-ahn 저도 같은문제때문에 고민하고 있는데 혹시 임시로 사용할 수 있는 방법이 있을까요? patch 라던지?
지금 당장 개발하고 있어서 시도하는데 현재로직상 지원하지 않는것 같아서요. ㅠㅠ
